python-threatexchange/threatexchange/signal_type/tests/test_pdq_index2.py

```diff
@@ -0,0 +1,195 @@
+import pytest
+import io
+import faiss
+from threatexchange.signal_type.pdq.pdq_index2 import (
+    PDQIndex2,
+    _PDQHashIndex,
+    DIMENSIONALITY,
+    DEFAULT_MATCH_DIST,
+)
+
+
+@pytest.fixture
+def empty_index():
+    """Fixture for an empty index."""
+    return PDQIndex2[str]()
+
+
+@pytest.fixture
+def custom_index_with_threshold():
+    """Fixture for an index with custom index and threshold."""
+    custom_index = faiss.IndexFlatL2(DIMENSIONALITY + 1)
+    custom_threshold = DEFAULT_MATCH_DIST + 1
+    return PDQIndex2[str](index=custom_index, threshold=custom_threshold)
+
+
+@pytest.fixture
+def sample_index():
+    """Fixture for an index with a small sample set."""
+    pdq_hashes = [
+        "f8f8f0cee0f4a84f06370a22038f63f0b36e2ed596621e1d33e6b39c4e9c9b22",
+        "f" * 64,
+        "0" * 64,
+        "a" * 64,
+    ]
+    index = PDQIndex2[str](entries=[(h, pdq_hashes.index(h)) for h in pdq_hashes])
+    return index, pdq_hashes
```

Comment:
blocking: I'm not sure about this pattern of returning an index and the sample hashes. It doesn't seem like you are saving that much time vs something like this:

```
SAMPLE_HASH = "f8f8f0cee0f4a84f06370a22038f63f0b36e2ed596621e1d33e6b39c4e9c9b22"

SAMPLE_HASHES = [
  SAMPLE_HASH,
  "f" * 64,
  "0" * 64,
  "a" * 64,
]

@dataclass
class IndexRecord:
  original_hash: string
  idx: int 

  def get_expected_distance(matched_hash) -> int:
    return pdq_utils.simple_distance_binary(self.original_hash, matched_hash)

def make_index(hashes):
  return PDQIndex2.build((h, StoredRecord(h, i) for i, h in enumerate(hashes)

def test_something():
  index = make_index(SAMPLE_HASHES)

  query_hash = "..."

  for d, record in index.search([query_hash ]):
    assert d == record.expected_distance(query_hash)
  
```

And you also aren't consistent on the test fixtures about what will be in the return. Fixtures are nice, but they aren't mandatory